app/editor/src/features/admin/sources/services/ScheduleAdvanced.tsx

```diff
@@ -39,9 +39,13 @@ export const ScheduleAdvanced: React.FC<IScheduleAdvancedProps> = (props) => {
   const handleDone = async (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
     if (index !== undefined) {
       try {
-        // TODO: Validation doesn't work.
-        await AdvancedScheduleSchema.validate(values.schedules[index], { abortEarly: false });
-        setIndex(undefined);
+        // Mock what formik does on submit, values must be "touched" for validation to fire.
+        const errors = await validateForm();
```

Comment:
I find the logic challenging here.  What will cause an error to be thrown here?  I would expect if validation to fail it would throw an error, which then would never result in `setTouched`.

Reply:
Doing a few other tweaks right now with the YupSchema